### DISCUSSION TITLE
Is there a way to use the provided ID instead of generate a custom one from useFieldArray()?

### DISCUSSION THREAD

**Discussion by BrunoQuaresma** [react-hook-form/react-hook-form#8850] [Q&A]
Hi folks, I have an array of objects with their own ID and I would like to use this ID instead of the ones generated by the `useFieldArray()`. Is there a way to disable the auto ID generation? Thanks!

**Comment by Moshyfawn** [comment]
Hey, @BrunoQuaresma!
 
No, RHF needs it's `id`s to track all the info about what the field is, it's position, etc.

While it's not as prominent when you simply update existing fields in an array, things get complicated when you start adding new fields: what ID should a new field have when you call `append(data)`?
